app/src/main/res/layout/activity_animal_details.xml

```diff
@@ -10,95 +9,105 @@
             type="tech.pauly.findapet.discover.AnimalDetailsViewModel" />
     </data>
 
-    <ScrollView
+    <android.support.design.widget.CoordinatorLayout
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:background="@android:color/white">
+        android:layout_height="match_parent">
 
-        <android.support.constraint.ConstraintLayout
+        <android.support.design.widget.AppBarLayout
+            android:id="@+id/app_bar_layout"
             android:layout_width="match_parent"
-            android:layout_height="wrap_content">
+            android:layout_height="400dp"
+            android:elevation="12dp">
 
-            <android.support.v7.widget.Toolbar
-                android:id="@+id/toolbar"
+            <android.support.design.widget.CollapsingToolbarLayout
                 android:layout_width="match_parent"
-                android:layout_height="?attr/actionBarSize"
-                android:background="@color/purpleStandard"
-                android:elevation="8dp"
-                android:theme="@style/ToolbarTheme"
-                app:layout_constraintTop_toTopOf="parent"
-                app:title="Pet Details"
-                app:titleTextAppearance="@style/FontStrongHeader"
-                app:titleTextColor="@android:color/white" />
-
-            <ImageView
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:layout_marginEnd="@dimen/standard_margin"
-                android:elevation="8dp"
-                android:src="@drawable/selector_favorite"
-                app:layout_constraintBottom_toBottomOf="@id/toolbar"
-                app:layout_constraintEnd_toEndOf="@id/toolbar"
-                app:layout_constraintTop_toTopOf="@id/toolbar" />
-
-            <ImageView
-                android:id="@+id/animal_image"
+                android:layout_height="match_parent"
+                app:contentScrim="?attr/colorPrimary"
+                app:layout_scrollFlags="scroll|exitUntilCollapsed|snap">
+
+                <FrameLayout
```

Comment:
Learned this one the hard way. Loading the image into Picasso and then having that view directly inside the AppBarLayout makes the image draw above the toolbar when collapsed. However, just putting it inside a simple FrameLayout fixes this. No reason why, I'm assuming something about draw order but can't figure out why this happens to Picasso but not a regular image.